editor/editor_settings.cpp

```diff
@@ -602,6 +602,10 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_RANGE, "filesystem/file_dialog/thumbnail_size", 64, "32,128,16")
 
 	// Quick Open dialog
+	EDITOR_SETTING_USAGE(Variant::INT, PROPERTY_HINT_RANGE, "filesystem/quick_open_dialog/max_results", 100, "0,10000,1", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_RESTART_IF_CHANGED)
```

Comment:
This doesn't need to require restart, but can be improved later.

Reply:
In the initial version of this PR, it wasn't required and instead there was some extra code to alloc more results but it had a bug (or more) so I backed it out. If you'd like though, the code is there if you can spot the issue.